src/profiler.cpp

```diff
@@ -784,6 +793,7 @@ void* Profiler::dlopen_hook(const char* filename, int flags) {
     void* result = dlopen(filename, flags);
     if (result != NULL) {
         instance()->updateSymbols(false);
+        MallocTracer::installHooks();
```

Comment:
While relying on a similar strategy, I recall we were not able to catch all loaded libraries. We eventually decided to put a timer that would trigger and recheck loaded libraries every X seconds (in addition to the dlopen hook). If this is of interest, I can find more details.

Reply:
It would be interesting to understand when this can happen.

Reply:
I found the commit thanks to @nsavoire. A private symbol can be used to dlopen `__libc_dlopen_mode`.
https://github.com/DataDog/ddprof/commit/d0bb6c663d98a0be891d8e5be8e7aea8067daa9a